portal-impl/src/com/liferay/portal/template/DefaultTemplateResourceLoader.java

```diff
@@ -128,9 +130,11 @@ public void clearCache(String templateId) {
 
 	@Override
 	public void destroy() {
-		_multiVMPool.removePortalCache(
+		PortalCacheHelperUtil.removePortalCache(
```

Comment:
Hi @marianoalvarosaiz,
Why you make this change? anything wrong you saw? do you have steps to reproduce?

Theoretically, we should not use the helper class here cause  the cache is retrieved directly, not through helper class, it should not be closed by helper class either.

Thanks.
Tina.

Reply:
Hi @marianoalvarosaiz ,
I know the reason you made this change, I will fix it in a general way.

Thanks.
Tina.

Reply:
Hi @tinatian,
 to reproduce it you just need:

1. Start the portal.
2. From the gogo console refresh the freemarker module.
3. You'll get an exception indicating the following: _The com.liferay.portal.kernel.template.TemplateResourceLoader.ftl Cache is not alive (STATUS_SHUTDOWN)_

The reason is because when you stop the module, and call FreeMarkerTemplateResourceLoader.destroy() to deactivate the component you destroy for both MultiVMEhcachePortalCacheManager and SingleVMEhcachePortalCacheManager the _com.liferay.portal.kernel.template.TemplateResourceLoader.ftl_

But aterwards, when you call FreeMarkerManager.doGetTemplate to retrieve the template, the helper is called from _com.liferay.portal.template.AbstractSingleResourceTemplate.getPortalCache(TemplateResource, String)_ and the DynamicPortalCacheManager still has the shutted down cache.

In one sentence, we're removing from the MultiVM/SingleVMEhcachePortalCacheManager but we're retrieving from DynamicPortalCacheManager that wraps the previous one but has the shutted down cache saved in _dynamicPortalCaches.

If need the generated exception is the following:

```java.lang.IllegalStateException: The com.liferay.portal.kernel.template.TemplateResourceLoader.ftl Cache is not alive (STATUS_SHUTDOWN)
	net.sf.ehcache.Cache$CacheStatus.checkAlive(Cache.java:4064)
	net.sf.ehcache.Cache.checkStatus(Cache.java:2755)
	net.sf.ehcache.Cache.get(Cache.java:1716)
	net.sf.ehcache.Cache.get(Cache.java:1696)
	com.liferay.portal.cache.ehcache.internal.EhcachePortalCache.doGet(EhcachePortalCache.java:83)
	com.liferay.portal.cache.BasePortalCache.get(BasePortalCache.java:39)
	com.liferay.portal.kernel.cache.DynamicPortalCache.get(DynamicPortalCache.java:51)
	com.liferay.portal.template.AbstractSingleResourceTemplate.cacheTemplateResource(AbstractSingleResourceTemplate.java:115)
	com.liferay.portal.template.AbstractSingleResourceTemplate.<init>(AbstractSingleResourceTemplate.java:57)
	com.liferay.portal.template.freemarker.internal.FreeMarkerTemplate.<init>(FreeMarkerTemplate.java:62)
	com.liferay.portal.template.freemarker.internal.FreeMarkerManager.doGetTemplate(FreeMarkerManager.java:359)
	com.liferay.portal.template.BaseSingleTemplateManager.getTemplate(BaseSingleTemplateManager.java:56)
	com.liferay.portal.template.BaseSingleTemplateManager.getTemplate(BaseSingleTemplateManager.java:48)
	com.liferay.portal.kernel.template.TemplateManagerUtil._getTemplate(TemplateManagerUtil.java:204)
	com.liferay.portal.kernel.template.TemplateManagerUtil.getTemplate(TemplateManagerUtil.java:80)
	com.liferay.taglib.util.ThemeUtil.doIncludeFTL(ThemeUtil.java:238)
	com.liferay.taglib.util.ThemeUtil.include(ThemeUtil.java:111)
	com.liferay.taglib.util.ThemeUtil.include(ThemeUtil.java:78)
	com.liferay.taglib.theme.IncludeTag.doEndTag(IncludeTag.java:34)
	org.apache.jsp.html.common.themes.portal_jsp._jspx_meth_liferay_002dtheme_005finclude_005f1(Unknown Source)
	org.apache.jsp.html.common.themes.portal_jsp._jspService(Unknown Source)
	org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
	javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
	com.liferay.portal.servlet.DirectRequestDispatcher.forward(DirectRequestDispatcher.java:49)
	com.liferay.portal.servlet.DirectRequestDispatcherFactoryImpl$IndirectRequestDispatcher.forward(DirectRequestDispatcherFactoryImpl.java:185)
	com.liferay.portal.servlet.ClassLoaderRequestDispatcherWrapper.doDispatch(ClassLoaderRequestDispatcherWrapper.java:82)
	com.liferay.portal.servlet.ClassLoaderRequestDispatcherWrapper.forward(ClassLoaderRequestDispatcherWrapper.java:45)
	com.liferay.portal.struts.StrutsUtil.forward(StrutsUtil.java:62)
	com.liferay.portal.struts.PortalRequestProcessor._internalModuleRelativeForward(PortalRequestProcessor.java:359)
	com.liferay.portal.struts.PortalRequestProcessor._processForward(PortalRequestProcessor.java:432)
	com.liferay.portal.struts.PortalRequestProcessor._process(PortalRequestProcessor.java:398)
	com.liferay.portal.struts.PortalRequestProcessor.process(PortalRequestProcessor.java:154)
	com.liferay.portal.internal.servlet.MainServlet.doGet(MainServlet.java:201)
	javax.servlet.http.HttpServlet.service(HttpServlet.java:634)
	com.liferay.portal.internal.servlet.MainServlet.service(MainServlet.java:620)```

Reply:
Hi @tinatian,
 I've just read your comment. Sorry, we crossed them. Ok then. Thanks a lot.
Kind regards.